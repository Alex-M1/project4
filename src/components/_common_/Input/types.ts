import { colors } from 'constants/colors';
import { SyntheticEvent } from 'react';

export interface IStInput {
  colors: typeof colors;
  theme: 'dark' | 'light';
}
export interface IInput {
  type: string;
  value: string;
  placeholder: string;
  onChange: (e: SyntheticEvent<HTMLInputElement>) => void;
}
