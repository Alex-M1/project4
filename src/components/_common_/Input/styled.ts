import styled from 'styled-components';
import { IStInput } from './types';

export const StInput = styled.input < IStInput > `
    width: 290px;
    height: 45px;
    border-radius: 10px;
    padding: 10px;
    outline: none;
    border: none;
    background: ${({ colors, theme }) => colors[theme].input};
    &::placeholder{
        color:${({ colors, theme }) => colors[theme].placeholderColor};
    }
`;
