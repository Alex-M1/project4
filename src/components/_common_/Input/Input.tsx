import React from 'react';
import { useTheme } from 'src/components/hooks/useTheme';
import { StInput } from './styled';
import { IInput } from './types';

const Input = ({ type, value, placeholder, onChange }: IInput) => {
  const { colors, theme } = useTheme();
  return (
    <StInput
      type={type === 'confirm' ? 'password' : type}
      value={value}
      theme={theme}
      colors={colors}
      onChange={onChange}
      placeholder={placeholder}
    />
  );
};

export default Input;
