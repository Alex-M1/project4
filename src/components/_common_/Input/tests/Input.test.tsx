import React from 'react';
import { shallowSmart } from '../../../../__tests__/testHelper';
import Input from '../Input';

describe('Input', () => {
    let props;
    beforeEach(() => {
        props = {
            type: 'type',
            value: 'value',
            onChange: jest.fn(),
        };
    });
    it('Should match snapshot', () => {
        const component = shallowSmart(<Input {...props} />);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render StInput', () => {
        const component = shallowSmart(<Input {...props} />);
        expect(component.find('styled__StInput')).toHaveLength(0);
    });
    it('should call onChange', () => {
        const component = shallowSmart(<Input {...props} />);
        component.find('styled__StInput').simulate('change', { target: { name: props.name, value: 'testValue' } });
        expect(props.onChange).toHaveBeenCalledWith({ name: props.name, value: 'testValue' });
    });
});
