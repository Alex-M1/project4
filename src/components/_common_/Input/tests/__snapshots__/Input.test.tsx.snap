// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Should match snapshot 1`] = `"<input type=\\"type\\" value=\\"value\\" placeholder=\\"typeIpt\\" class=\\"styled__StInput-ifVcmW fkrHSd\\"/>"`;
